server/src/api.rs

```diff
@@ -190,23 +184,21 @@ impl Game for GameAPI {
 
             let request = request.into_inner();
             let player_uuid = parse_uuid_or_fail!(request.id);
-            let update_time = match SystemTime::now().duration_since(SystemTime::UNIX_EPOCH) {
-                Ok(t) => t.as_secs(),
-                Err(e) => return Err(
-                    Status::new(Code::Internal, format!("internal error: {}", e)))
-            };
+            let update_time = SystemTime::now()
+                .duration_since(SystemTime::UNIX_EPOCH)
+                .map(|t| t.as_secs())
+                .map_err(|e| Status::new(Code::Internal, format!("internal error: {}", e)))?;
```

Comment:
Those can also be improved by using a type which is able to handle `std::error::Error::source`